cluster/src/main/java/com/hartwig/pipeline/storage/StorageProvider.java

```diff
@@ -16,7 +17,15 @@ private StorageProvider(final Arguments arguments, final GoogleCredentials crede
     }
 
     public Storage get() {
-        return StorageOptions.newBuilder().setCredentials(credentials).setProjectId(arguments.project()).build().getService();
+        StorageOptions.Builder builder = StorageOptions.newBuilder();
+        return builder.setCredentials(credentials)
+                .setProjectId(arguments.project())
+                .setTransportOptions(HttpTransportOptions.newBuilder()
+                        .setConnectTimeout(Integer.MAX_VALUE)
+                        .setReadTimeout(Integer.MAX_VALUE)
```

Comment:
I think you can pass "0" here if you want an infinite timeout.

[reference](https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/google-cloud-core-http/src/main/java/com/google/cloud/http/HttpTransportOptions.java)

Reply:
Good idea